src/SonarScanner.MSBuild.TFS/CoverageReportProcessorBase.cs

```diff
@@ -110,18 +110,25 @@ public bool ProcessCoverageReports()
 
         private bool TryConvertCoverageReports(IEnumerable<string> vscoverageFilePaths, out IEnumerable<string> vscoveragexmlPaths)
         {
-            var xmlFileNames = vscoverageFilePaths.Select(x => Path.ChangeExtension(x, XmlReportFileExtension));
+            var xmlFileNames = new List<string>();
 
-            Debug.Assert(!xmlFileNames.Any(x => File.Exists(x)),
-                "Not expecting a file with the name of the binary-to-XML conversion output to already exist.");
-
-            var anyFailedConversion = vscoverageFilePaths.Zip(xmlFileNames, (x, y) => new { coverage = x, xml = y })
-                .Any(x => !this.converter.ConvertToXml(x.coverage, x.xml));
-
-            if (anyFailedConversion)
+            foreach (var vscoverageFilePath in vscoverageFilePaths)
             {
-                vscoveragexmlPaths = Enumerable.Empty<string>();
-                return false;
+                var xmlFilePath = Path.ChangeExtension(vscoverageFilePath, XmlReportFileExtension);
+                if(File.Exists(xmlFilePath))
+                {
+                    this.Logger.LogInfo($"Found corresponding Binary-to-XML conversion output file for {vscoverageFilePath}, no conversion will be tried.");
```

Comment:
This string should be in a resource file, and it would read slightly better if it said "no conversion will be attempted."